packages/composites/token-allocation/test/integration/adapter.test.ts

```diff
@@ -1,34 +1,151 @@
-import { Requester } from '@chainlink/ea-bootstrap'
-import { assertError } from '@chainlink/ea-test-helpers'
-import { AdapterRequest } from '@chainlink/types'
-import { makeExecute } from '../../src/adapter'
-import { makeConfig } from '../../src/config'
+import { AdapterRequest, Execute } from '@chainlink/types'
+import * as tokenAllocationAdapter from "../../src/index" 
+import { Requester } from "@chainlink/ea-bootstrap"
 
-describe('execute', () => {
-  const jobID = '1'
-  process.env.DATA_PROVIDER_URL = 'ignoreable'
-  const execute = makeExecute(makeConfig(''))
-
-  describe('error calls @integration', () => {
-    const requests = [
-      {
-        name: 'invalid units',
-        testData: {
-          id: jobID,
-          data: { allocations: [{ symbol: 'DAI', balance: '1000000000000000000', decimals: 18 }] },
+const jobRunID = 1
+const sources = {
+    "amberdata": {
+        providerUrlEnvVar: `AMBERDATA_DATA_PROVIDER_URL`
+    }, 
+    "coinapi": {
+        providerUrlEnvVar: `COINAPI_DATA_PROVIDER_URL`
+    }, 
+    "coingecko": {
+        providerUrlEnvVar: `COINGECKO_DATA_PROVIDER_URL`,
+        isBatched: true 
+    }, 
+    "coinmarketcap": {
+        providerUrlEnvVar: `COINMARKETCAP_DATA_PROVIDER_URL`,
+        isBatched: true
+    }, 
+    "coinpaprika": {
+        providerUrlEnvVar: `COINPAPRIKA_DATA_PROVIDER_URL`
+    }, 
+    "cryptocompare": {
+        providerUrlEnvVar: `CRYPTOCOMPARE_DATA_PROVIDER_URL`,
+        isBatched: true
+    }, 
+    "kaiko": {
+        providerUrlEnvVar: `KAIKO_DATA_PROVIDER_URL`
+    }, 
+    "nomics": {
+        providerUrlEnvVar: `NOMICS_DATA_PROVIDER_URL`,
+        isBatched: true
+    }, 
+    "tiingo": {
+        providerUrlEnvVar: `TIINGO_DATA_PROVIDER_URL`
+    }
+}
+Requester.request = (request: any) => {
+    const source = sources[request.baseURL]
+    const { quote } = request.data.data 
+    if(source.isBatched || (request.baseURL === "coinapi" && request.data.data.quote === "USD")) {
```

Comment:
We should use `nock` instead here if possible, that way we dont need to mock out `Requester.request`